src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FixedAssetsInvestmentsService.java

```diff
@@ -0,0 +1,194 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.GenerateEtagUtil;
+import uk.gov.companieshouse.api.accounts.Kind;
+import uk.gov.companieshouse.api.accounts.ResourceName;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.links.BasicLinkType;
+import uk.gov.companieshouse.api.accounts.links.SmallFullLinkType;
+import uk.gov.companieshouse.api.accounts.links.TransactionLinkType;
+import uk.gov.companieshouse.api.accounts.model.entity.notes.fixedassetsinvestments.FixedAssetsInvestmentsEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.fixedassetsinvestments.FixedAssetsInvestments;
+import uk.gov.companieshouse.api.accounts.repository.FixedAssetsInvestmentsRepository;
+import uk.gov.companieshouse.api.accounts.service.ResourceService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseStatus;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transformer.FixedAssetsInvestmentsTransformer;
+import uk.gov.companieshouse.api.accounts.utility.impl.KeyIdGenerator;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Service
+public class FixedAssetsInvestmentsService implements ResourceService<FixedAssetsInvestments> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    private FixedAssetsInvestmentsRepository repository;
+    private FixedAssetsInvestmentsTransformer transformer;
+    private KeyIdGenerator keyIdGenerator;
+    private SmallFullService smallFullService;
+
+    @Autowired
+    public FixedAssetsInvestmentsService(FixedAssetsInvestmentsRepository repository,
+            FixedAssetsInvestmentsTransformer transformer,
+            KeyIdGenerator keyIdGenerator,
+            SmallFullService smallFullService) {
+
+        this.repository = repository;
+        this.transformer = transformer;
+        this.keyIdGenerator = keyIdGenerator;
+        this.smallFullService = smallFullService;
+    }
+
+    @Override
+    public ResponseObject<FixedAssetsInvestments> create(FixedAssetsInvestments rest,
+            Transaction transaction,
+            String companyAccountId,
+            HttpServletRequest request) throws DataException {
+
+        setMetadataOnRestObject(rest, transaction, companyAccountId);
+
+        FixedAssetsInvestmentsEntity entity = transformer.transform(rest);
+        entity.setId(generateID(companyAccountId));
+
+        try {
+            repository.insert(entity);
+        } catch (DuplicateKeyException e) {
+            LOGGER.errorRequest(request, e, getDebugMap(transaction, companyAccountId,
+                entity.getId()));
```

Comment:
the services have all been updated to no longer LOG the exceptions in the service layer. So can be removed and the logging is handled within the controller.

Reply:
Services, controller and tests updated to match the tidier logging approach that is in develop